File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        int maxLen = 0;
        var dp = new HashMap<Integer, Integer>();
        for(int num : nums) {
            dp.put(num, 1 + dp.getOrDefault(num, 0));
            if(dp.containsKey(num - 1))
                maxLen = Math.max(maxLen, dp.get(num) + dp.get(num - 1));
            if(dp.containsKey(num + 1))
                maxLen = Math.max(maxLen, dp.get(num) + dp.get(num + 1));
        }
        return maxLen;
    }
}

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        int maxLen = 0;
        unordered_map<int, int> dp;
        for(int &num : nums) {
            dp[num]++;
            if(dp.count(num - 1)) maxLen = max(maxLen, dp[num] + dp[num - 1]);
            if(dp.count(num + 1)) maxLen = max(maxLen, dp[num] + dp[num + 1]);
        }
        return maxLen;
    }
};